### ISSUE TITLE
some jetty versions cause KWs to be executed twice

### ISSUE THREAD

**Issue by karniemi** [robotframework/jrobotremoteserver#39]
Yep,

we had a long time head-ache: occasionally, the KWs were getting executed twice in parallel. We only noticed the root of our headache after adding more logging in our java KWs, and finally added the java-thread-number for the logs. We noticed that sometimes two threads are executing our KWs in parallel.  We are running UI testing with selenium via jrobotremoteserver...and occasionally there were 2threads clicking the UI at the same time...which is not nice :-).

We found out that there have been reports against jetty for closing the persistent connection in a way that violates specifications(don't ask for reference, it's a long time).

Also in python 2.7 xmlrpclib.py, there's a retry loop https://github.com/python/cpython/blob/2.7/Lib/xmlrpclib.py
``` for i in (0, 1):```

In some python3 ticket, "they" referred the code in that for loop to be an example of badly behaving http user. The exception/connection handing was improved, but only for the corresponding python3 lib. 

Long story short: the pain is gone after overriding jetty versions in our pom.xml to "org.eclipse.jetty:jetty-servlet:8.1.22.v20160922" and "org.eclipse.jetty:jetty-http:8.1.22.v20160922". Still running robot-framework with python2.7 though.

Just wanted to inform about this if someone's having the same pain...

**Comment by ombre42** [comment]
Thanks for the info. I have definitely seen some strange behavior with regards to how connection life cycle.

**Comment by karniemi** [comment]
Now, 3 months later, CI tests running day and night in several labs...we can say for sure that our issues were resolved by updating the jetty versions.

**Comment by Hi-Fi** [comment]
Jetty is now (with version 4) updated to version 9, so this is not probably valid anymore.
